doc/tutorials/troubleshooting.rst

```diff
@@ -0,0 +1,47 @@
+Troubleshooting
+===============
+
+Plugin errors
+-------------
+
+Errors that start with ``GHC Core to PLC plugin`` are ``plutus-tx-plugin`` errors.
+The most common error is:
+
+``Error: Reference to a name which is not a local, a builtin, or an external INLINABLE function``
+
+This means the plugin doesn't have access to implementation of the function, which it needs to be able to compile the function to Plutus Core.
+Some things you can do to fix it:
+
+- Make sure to add ``{-# INLINABLE functionname #-}`` to your function.
+- If there's an extra ``$c`` in front of the function name in the error, GHC has generated a specialised version of your function,
+  which prevents the plugin from accessing it.
+  You can turn it off with ``{-# OPTIONS_GHC -fno-specialise #-}``
+- Other compiler options that could fix the issue are
+
+  - ``{-# OPTIONS_GHC -fno-strictness #-}``
+  - ``{-# OPTIONS_GHC -fno-ignore-interface-pragmas #-}``
+  - ``{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}``
+  - ``{-# OPTIONS_GHC -fobject-code #-}``
+
+  Some more details are in `the plutus-tx readme <https://github.com/input-output-hk/plutus/tree/master/plutus-tx#building-projects-with-plutus-tx>`_.
+
+If you don't need the plugin to succeed, f.e. when Haddock is building documentation,
+you can pass the flag ``-fdefer-plugin-errors``.
```

Comment:
This isn't quite right: `defer-plugin-errors` is a *Cabal* flag that we declare in our packages. Setting it then sets the actual GHC flag `-fplugin-opt PlutusTx.Plugin:defer-errors` that does the deferring. So for users we should recommend they set the latter as a GHC option.

(Why do we do it this way? because it's easier to expose it as a cabal flag for other tooling)